kms/api/CloudKmsTest/CloudKmsTest.cs

```diff
@@ -279,6 +294,124 @@ private bool AssertRoleAndMemberPresent(string response, string role, string mem
+            CryptoKey cryptoKeyToCreate = new CryptoKey
+            {
+                Purpose = purpose,
+                VersionTemplate = new CryptoKeyVersionTemplate
+                {
+                    Algorithm = algorithm,
+                    ProtectionLevel = ProtectionLevel.Software,
+                },
+            };
+            CryptoKey key = client.CreateCryptoKey(keyRingName, cryptoKey, cryptoKeyToCreate);
+
+            CryptoKeyVersionName ckvName = CryptoKeyVersionName.Parse(key.Name + "/cryptoKeyVersions/1");
+            CryptoKeyVersion ckv = client.GetCryptoKeyVersion(ckvName);
+
+            // The version is generated asynchronously. Wait for it to be in state Enabled before
+            // proceeding. Throw an exception if it hasn't been generated after 5 minutes.
+            Task refresher = Task.Run(async () =>
+            {
+                while (ckv.State != CryptoKeyVersion.Types.CryptoKeyVersionState.Enabled)
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(5));
+                    ckv = client.GetCryptoKeyVersion(ckvName);
+                }
+            });
+            if (!refresher.Wait(TimeSpan.FromMinutes(5)))
+            {
+                throw new TimeoutException($"Timed out waiting for version '{{ckv.Name}}' to become enabled.");
+            }
+
+            return ckvName;
+        }
+
+        [Fact]
+        public void TestAsymmetricSignAndVerify()
+        {
+            string contentFile = @"../../../../test/data/test_file.txt";
+            string signatureFile = @"../../../../test/data/test_file_signature.bin";
+            string timeStamp = $"-{DateTime.Now.ToString("yyyyMMddHHmmssfff")}";
+            string keyRing = $"testKeyRing{timeStamp}";
+            string cryptoKey = $"testCryptoKey{timeStamp}";
+            KeyRingName keyRingName = new KeyRingName(s_projectId, "global", keyRing);
+
+            Run("createKeyRing", keyRingName.ProjectId, keyRingName.LocationId, keyRingName.KeyRingId);
+
+            CryptoKeyVersionName version = BuildAsymmetricKey(
+                keyRingName, cryptoKey, CryptoKey.Types.CryptoKeyPurpose.AsymmetricSign,
+                CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm.EcSignP256Sha256);
+
+            var signOutput = Run("asymmetricSign", version.ProjectId, version.LocationId, version.KeyRingId,
+                    version.CryptoKeyId, version.CryptoKeyVersionId, contentFile, signatureFile);
+            Assert.Equal(0, signOutput.ExitCode);
+
+            var verifyOutput = Run("asymmetricVerify", version.ProjectId, version.LocationId, version.KeyRingId,
+                    version.CryptoKeyId, version.CryptoKeyVersionId, contentFile, signatureFile);
+            Assert.Equal(0, verifyOutput.ExitCode);
+            Assert.Contains("verified: True", verifyOutput.Stdout);
+
+            // Delete signature file to clean up
+            File.Delete(signatureFile);

```

Comment:
Shouldn't you delete the key you created server side as well?

Reply:
What is the concern? Billing?

This is consistent with the tests that are already here.

Reply:
Quota, which I'm not sure if it applies for this API, but we have other samples failing because they hit quota all the time because there's no cleanup. Leave as is if the rest is like that but this will change as well in the near future.